src/nvim/CMakeLists.txt

```diff
@@ -189,8 +192,14 @@ list(APPEND NVIM_LINK_LIBRARIES
     ${LIBUV_LIBRARIES}
     ${MSGPACK_LIBRARIES}
     ${LUAJIT_LIBRARIES}
+    "-Wl,--whole-archive"
```

Comment:
It's because linking is very linear.  In this case, libtickit depends on both libtermkey and libunibilium.  If you link libtickit last, then those functions won't be resolved again (because they were present in previous libraries).  However, if you change the order so that libtickit gets linked first, then the linker knows about the unresolved functions in libtickit can fix them up correctly when libtermkey and libunibilium are linked in next.

In general, you want the objects with the most dependencies to be linked in first, and the objects with no dependencies linked in last.  And you want to make sure to link the dependencies in after the object that requires them.  What you are really doing is a topological sort. :-)